### ISSUE TITLE
[Bug][ibm_zos_core] When using the release branch 1.5.0 with python 3.10 or later and pipelining=true fails

### ISSUE THREAD

**Issue by ddimatos** [ansible-collections/ibm_zos_core#570]
### Bug description

This is a fairly complex issue in that it is not obvious to most because:
- ibm_zos_core 1.5.0-beta.1 will be the first version to officially support python 3.10 and 3.11 thus many have not tried this combination yet.
- The problem only surfaces when using IBM Enterprise Python 3.10 or later
- The other reason is that this problem is not apparent is because in Ansible 2.11 and later the `ansible.cfg` key `pipelining` has been moved under section `[connection]` where prior it was under `[ssh_connection]` so if one does not update their configuration `pipelining` defaults to `False` and there is no problem with the collection when its false.
- The other thing to note is that this is not limited to `ibm_zos_core` this impacts any of the community modules used to communicate with a z/OS managed node with `pipelining` = `True`

This happens only when using the pipelining feature with Python 3.10 or later and the default z/OS Shell. It has been determined this is happening because the pipelining pipes directly into Pythons STDIN and **the pipe used in z/OS is not tagged to be UTF-8** and appears to be EBCDIC which then confuses python and its not able to correctly operate because Python when it sees an untagged file it treats it as binary and does nothing with the encoding which then breaks the interpreter. 

Using the bash shell is a work around but bash is not a supported shell and there is no plan in site to support the bash shell. You can enable the shell with `ansible_python_interpreter` and then pipelining should work.

Some further debugging show the behavior in a more controlled space:
```
# echo "$SHELL ($0)"; echo "print('Testing print message')" | /allpython/3.10/usr/lpp/IBM/cyp/v3r10/pyz/bin/./python
/bin/sh (-sh)
  File "<stdin>", line 1
    �����M}ㅢ����@�����@�������}]
          ^
SyntaxError: unmatched '}'
```

```
bash-4.3# echo "$SHELL ($0)"; echo "print('Testing print message')" | /allpython/3.10/usr/lpp/IBM/cyp/v3r10/pyz/bin/./python
/bin/sh (/u/oeusr01/tools/bash)
Testing print message
```

Plan:
- open an RFE to tag the pipe in z/OS shell
- review the ansile ssh code to see if we can mock up an edit to set the encoding=UTF-8 in the subproccess PIPE
- Consider providing a patch that tweaks the ssh commnd
- Work with the Python team to absorb this detection to get to market faster. 

### Playbook verbosity output

Python 3.10.5
```
fatal: [zvm]: FAILED! => {"changed": false, "module_stderr": "SyntaxError: Non-UTF-8 code starting with '\\xa4' in file <stdin> on line 2, but no encoding declared; see https://python.org/dev/peps/pep-0263/ for details\n", "module_stdout": "", "msg": "MODULE FAILURE\nSee stdout/stderr for the exact error", "rc": 1}
```

Python 3.10
```
�����m�������K���}]\u0015@@@@@@@@����@����M������m���k@}��}]@��@������z\u0015@@@@@@@@@@@@������K�����M������K��􄅃���M�������]]\u0015@@@@@@@@��@���M���K����]@~~@�z\u0015@@@@@@@@@@@@��������@~@�����M���K������k@������m���k@���������m������]\u0015@@@@@@@@����z\u0015@@@@@@@@@@@@������m������M������m���k@����m����k@���������m������]\u0015@@@@�������z\u0015@@@@@@@@���z\u0015@@@@@@@@@@@@������K������M����m����]\u0015@@@@@@@@������@MՁ��ř���k@��ř���]z\u0015@@@@@@@@@@@@����\u0015@@@@���K����M��������]\u0015��@mm����mm@~~@}mm����mm}z\u0015@@@@m���������m����M]\n                                                             ^\nSyntaxError: (unicode error) 'utf-8' codec can't decode byte 0x81 in position 2: invalid start byte\n", "module_stdout": "", "msg": "MODULE FAILURE\nSee stdout/stderr for the exact error", "rc": 1}
```

### Contents of `ansible.cfg`

```YAML
Env vars used

  _BPXK_AUTOCVT: "ON"
  LIBPATH: "{{ PYZ }}/lib:/lib:/usr/lib:."
  PATH: "{{ PYZ }}/bin:/bin:/var/bin"
  _CEE_RUNOPTS: "FILETAG(AUTOCVT,AUTOTAG) POSIX(ON)"
  _TAG_REDIR_ERR: "txt"
  _TAG_REDIR_IN: "txt"
  _TAG_REDIR_OUT: "txt"
  LANG: "C"
```
```


### Contents of the inventory

_No response_

### Contents of `group_vars` or `host_vars`

_No response_

### Ansible version

```shell
2.11.12
```

### IBM z/OS Ansible core Version

v1.4.0-beta.2

### IBM ZOAU version

v1.1.1

### z/OS version

2.4

**Comment by ddimatos** [comment]
Similar issues can be `UnicodeDecodeError: ‘utf-8’ codec can’t decode byte 0x84 in position 2:` and such.

**Comment by ddimatos** [comment]
Evaluating python patch.

**Comment by ddimatos** [comment]
Latest Python PTF due Mid March resolves this, more this after the PTF is released.

**Comment by ddimatos** [comment]
Will use this to evaluate if any doc needs to be done on the collection, likely all doc related work will happen in the playbook repository where samples are provided and/or the doc site.

**Comment by ddimatos** [comment]
Being tracked in the repo: https://github.com/IBM/z_ansible_collections_samples with issue https://github.com/IBM/z_ansible_collections_samples/issues/204 under PR https://github.com/IBM/z_ansible_collections_samples/pull/206

**Comment by staoli** [comment]
We recently upgraded from Python 3.9 to Python 3.11 and are facing this issue. You mentioned a PTF due mid March. Is this PTF available in the meantime? 
And do I understand correctly that unless we install the PTF the only workaround is using Bash instead of default z/OS shell (or going back to Python 3.9)?

**Comment by ddimatos** [comment]
@staoli - hi thanks for inquiring. 
Yes the PTF is available for Python 3.10 and 3.11, those notices can be found [here](https://www.ibm.com/support/pages/fixlist-ibm-open-enterprise-python-zos).

For 3.10:
PTF: UI91082
APAR: PH53438

For 3.11:
PTF: UI91094
APAR: PH53470

The new environment variable  `PYTHONSTDINENCODING: "cp1047"` has been included in all our [samples](https://github.com/IBM/z_ansible_collections_samples) but also can be seen [here](https://github.com/ansible-collections/ibm_zos_core/discussions/657).

This is only to allow for when `ansible.cfg` is configured for `pipelining=true` otherwise without the APAR/PTF described here, you should set `pipelining=false`. 

Should have `PYTHONSTDINENCODING: "cp1047"` in your environment variables without the APAR/PTF applied, in my testing it was ignored.

You have another choice besides going back to Python 3.9 or choosing `bash`, you can disable set your `pipelining=false`, performance will be impacted though.

I can not recommend `bash` as it is not a supported shell for our collection. Early on, we had many problems with it, although for this particular issue; tagging of pipes, it is actually performing as we would expect from a shell. My mention of it here was aid the issue with another data point, so that I can explore a permanent solution with Unix System Services while the Python team absorbed the issue in the short term. 

I will have you note, we perform no testing on the `bash` shell, in your usage, you find it performs well enough to use it. 

Let us know if we can help any further.
